### PR TITLE
fix: ensure nestd allOfs are correctly resolved

### LINKED ISSUE #94: allOf with deep relationships does not work consistently
The oneOf resolution depends on the sort order of the entities to work properly.

This hierarchy works ok because they are in sort order: `UserEntity -> UpdatableEntity -> BaseEntity`, but this one fails: `OrgEntity -> UpdatableEntity -> BaseEntity`.

I think it is a defect to depend on the sort order for this to work.

The fix is to add `right = resolveAllOf(right, passed)` next to: 
 https://github.com/contiamo/openapi-generator-go/blob/56c67ab16713c7ade8352748f772b775704cf86b/pkg/generators/models/models.go#L448

But that breaks a test case.

### PR DESCRIPTION
**Pull request by LucasRoesler**
Adjust the schema merge logic to include allOf resolution. This ensures that nested allOf's are correctly merged together.

Adjust some of the model properties generation to handle some edge cases that were discovered for singleton allOfs.

Resolves #94

### CODE REVIEW

File: pkg/generators/models/models_test.go [hunk 1]
```diff
@@ -214,9 +218,9 @@ func TestModels(t *testing.T) {
 }
 
 func TestModelsSingleCase(t *testing.T) {
-	t.Skip("only used during local development")
+	// t.Skip("only used during local development")
```
Comment (megaflo) [inline pkg/generators/models/models_test.go:221]: Is this an artifact from local testing?

Comment (LucasRoesler) [reply pkg/generators/models/models_test.go]: updated

File: pkg/generators/models/testdata/cases/allOf_enum_merging_and_validation/expected/model_connection.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func (m Connection) Validate() error {
 			m.Properties, validation.NotNil,
 		),
 		"technology": validation.Validate(
-			m.Technology, validation.NotNil,
+			m.Technology, validation.Required,
```
Comment (LucasRoesler) [inline pkg/generators/models/testdata/cases/allOf_enum_merging_and_validation/expected/model_connection.go:50]: hmm i just realized that these changed ... i need to double check that

Comment (injeniero) [reply pkg/generators/models/testdata/cases/allOf_enum_merging_and_validation/expected/model_connection.go:50]: I think that is the correct value given: https://github.com/contiamo/openapi-generator-go/pull/95/files/2e92689f7076065c27e9340b14de37bb780bfce9#diff-f6646d69f3c9d3794d9fde9eec4b73f1703ac88997bc53e93232810b6a5cd7f1L15

**Comment by injeniero** [comment]
Maybe for another PR, but I'm having issues with the dependency: `github.com/getkin/kin-openapi v0.89.0`, the current version is `v0.115.0` and it does fix a lot of bugs, one of them is to properly resolve external dependencies. I tried updating it here, but it has a few breaking changes, nothing really complicated to fix.

**Comment by injeniero** [comment]
I just tested it with my full schema and it works perfect!! Thanks!!

**Comment by injeniero** [comment]
@LucasRoesler what else is needed to merge?

**Comment by LucasRoesler** [comment]
Nothing is blocking it, i was just traveling for a few days. Merging now.

**Pull request merged**